### ISSUE TITLE
http协议不能创建修改数据，bolt协议可以，这是什么引起的

### ISSUE THREAD

**Issue by RickLwy** [neo4j-php/neo4j-php-client#49]

**Comment by transistive** [comment]
Sorry I do not speak chinese. If you have trouble with the HTTP drivers, please make sure you have installed a PSR HTTP client and factory: 
`composer require nyholm/psr7 nyholm/psr7-server kriswallsmith/buzz`

**Comment by RickLwy** [comment]
> Sorry I do not speak chinese. If you have trouble with the HTTP drivers, please make sure you have installed a PSR HTTP client and factory:
> `composer require nyholm/psr7 nyholm/psr7-server kriswallsmith/buzz`

谢谢！ 这个我已经引入了。可以返回数据，但是节点没有显示，也搜索不到
![image](https://user-images.githubusercontent.com/25677721/120980583-54245600-c7a9-11eb-84d3-567aa33ab027.png)

**Comment by transistive** [comment]
If I look at your example it seems to be working as intended. Are there any more problems?

**Comment by RickLwy** [comment]
创建的数据，查找不到。浏览器中也不显示

**Comment by transistive** [comment]
But when looking at the output of `print_r($result)` it is there. You can access it like this: `$node = $res->first()->get('n')`. The node is an array so you can access the name like this: `$node['name']`


We are working hard on version 2.0 at the moment to return a Node object instead of an array.

**Comment by RickLwy** [comment]
I'm sorry you might have misunderstood what I mean

The data returned is OK。The data I created is name ='bbbb ', but I can't find it again

Thank you for your answer

**Comment by transistive** [comment]
No problem!

You can work with data like this:

```php
$results = $client->run('MATCH (n:zzz {n.name: $name}) RETURN n', ['name' => 'bbbb']); // returns all result rows
$row = $results->first(); // returns the first row of the results (same as $results->get(0) or $results[0])
$node = $row->get('n'); // returns the node as an array
$name = $node['name']; // Accesses the name attribute of the node
```

Does this example help?

**Comment by RickLwy** [comment]
![image](https://user-images.githubusercontent.com/25677721/120987711-6eadfd80-c7b0-11eb-90d7-9e2caa367ae8.png)

Data not found for bbbb

**Comment by transistive** [comment]
When looking at the screenshot it appears to have a name 'aaaa' not 'bbbb' so you have to change the first line to this:

`$results = $client->run('MATCH (n:zzz {n.name: $name}) RETURN n', ['name' => 'aaaa']);`

**Comment by RickLwy** [comment]
不是这个意思。
1：create(n:zzz) set n.name="bbbb" return n
2：match (n:zzz) where n.name="bbbb" return n
第二步，http协议不能查到数据。bolt协议可以查到

**Comment by transistive** [comment]
I am able to reproduce the issue. I will take a look at it now. Thank you for bringing this to my attention.

**Comment by RickLwy** [comment]
Thank you, too

**Comment by transistive** [comment]
I have fixed the issue! It is available in version 1.3.3. If there are any other issues that come up, please notify me.

Thank you for your patience.

**Comment by RickLwy** [comment]
![image](https://user-images.githubusercontent.com/25677721/121129061-f5b9af00-c85e-11eb-8518-8013624e0749.png)
使用事务（Transaction）的时候，http协议不能创建数据。bolt协议可以创建

**Comment by transistive** [comment]
Thank you RickLwy!

I have immediately fixed the issue. It is now available in version 1.3.4

Please do not hesitate if you find more issues :+1:

**Comment by RickLwy** [comment]
![image](https://user-images.githubusercontent.com/25677721/121282721-8f8b6580-c90c-11eb-95bf-2d0d588ba784.png)
http协议只能保存`ab`不能保存`cd` 。bolt协议可以保存`abcd`。

**Comment by transistive** [comment]
Thans again! It is now fixed in version 1.3.5
